src/xcoff.rs

```diff
@@ -498,6 +498,48 @@ pub const C_EFCN: u8 = 255;
 /// Reserved.
 pub const C_TCSYM: u8 = 134;
 
+/// Csect auxiliary entry for C_EXT, C_WEAKEXT, and C_HIDEXT symbols.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct CsectAux32 {
+    /// Section length.
+    pub x_scnlen: U32<BE>,
+    /// Offset of parameter type-check hash in .typchk section.
+    pub x_parmhash: U32<BE>,
+    /// .typchk section number.
+    pub x_snhash: U16<BE>,
+    /// Symbol alignment and type.
+    pub x_smtyp: u8,
+    /// Storage mapping class.
+    pub x_smclas: u8,
+    /// Reserved.
+    pub x_stab: U32<BE>,
+    /// x_snstab.
+    pub x_snstab: U16<BE>
+}
+
+/// Csect auxiliary entry for C_EXT, C_WEAKEXT, and C_HIDEXT symbols.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct CsectAux64 {
+    /// Low 4 bytes of section length.
+    pub x_scnlen_lo: U32<BE>,
+    /// Offset of parameter type-check hash in .typchk section.
+    pub x_parmhash: U32<BE>,
```

Comment:
I see such fields are `_OFFSET32` type in headers:

```c
/* Define macros for file or section offsets and section sizes.
 * By default, many XCOFF structure fields are signed types.
 * By defining _XCOFF_UOFFSET, unsigned types are used.
 */
#ifdef _XCOFF_UOFFSET
#define _OFFSET32   unsigned int
#define _OFFSET32i  unsigned int
#define _INT32      unsigned int
#define _OFFSET64   unsigned long long
#else
#define _OFFSET32   _LONG32
#define _OFFSET32i  int
#define _INT32      int
#define _OFFSET64   long long
#endif
```

Is `_XCOFF_UOFFSET` defined in practice?

Reply:
Current definition on signed/unsigned type is not that strict. We define most fields as unsigned type except for some fields that must be defined as signed type in practice, which is consistent with the code behavior of other file formats.